packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -143,6 +144,76 @@ void main() {
     },
   );
 
+  testWidgets('iOS cursor has offset', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const CupertinoApp(
+        home: CupertinoTextField(),
+      ),
+    );
+
+    final EditableText editableText = tester.firstWidget(find.byType(EditableText));
+    expect(editableText.cursorOffset, const Offset(-2.0 / 3.0, 0));
+  });
+
+  testWidgets('Cursor animates on iOS', (WidgetTester tester) async {
+    debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+    await tester.pumpWidget(
+      const CupertinoApp(
+        home: CupertinoTextField(),
+      ),
+    );
+
+    final Finder textFinder = find.byType(CupertinoTextField);
+    await tester.tap(textFinder);
+    await tester.pump();
+
+    final EditableTextState editableTextState = tester.firstState(find.byType(EditableText));
+    final RenderEditable renderEditable = editableTextState.renderEditable;
+
+    expect(renderEditable.cursorColor.alpha, 255);
+
+    await tester.pump(const Duration(milliseconds: 100));
```

Comment:
what are all these for?

Reply:
Because the animation needs multiple 'frames' for the tests to work correctly. But it really only needs two. Should I condense these?

Reply:
Yes, it should only need to pump each time you trigger an animateTo on the animation controller via some time elapse. Reduce it to what's needed for the test to pass and add comments in the various places that does more than one pump.

Reply:
ok